### PR TITLE
Chronos: support tsdataset for validation_data in forecasters' fit

### PR DESCRIPTION
**Pull request by Chaselzxy**
## Description
At present in forecasters' fit, validation data supports the numpy ndarray and pytorch dataloader.

### 1. Why the change?

Support tsdataset for validation_data in forecasters' fit.

### 3. Summary of the change 

Changed `fit` in `base_forecaster`.
And changed unit test files for 4 forecasters: lstm, s2s, tcn and nbeats.

### 4. How to test?
- [ ] Unit test

### CODE REVIEW

**Comment by TheaperDeng** [comment]
@liangs6212 please also have a look, thx.

File: python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py [hunk 1]
```diff
@@ -392,6 +399,16 @@ def fit(self, data, validation_data=None, epochs=1, batch_size=32, validation_mo
...
+                if isinstance(validation_data, TSDataset):
+                    _rolled = validation_data.numpy_x is None
+                    validation_data =\
+                    validation_data.to_torch_data_loader(batch_size=batch_size,
+                                                         roll=_rolled,
+                                                         lookback=self.data_config['past_seq_len'],
+                                                         horizon=self.data_config['future_seq_len'],
+                                                         feature_col=validation_data.roll_feature,
+                                                         target_col=validation_data.roll_target,
+                                                         shuffle=True)
```
Comment (liangs6212) [inline python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py:411]: shuffle=False
btw, only the training dataset needs to be shuffled.

**Comment by TheaperDeng** [comment]
will take over this PR by myself
